modules/contracts/apps/challenge/src/state.rs

```diff
@@ -0,0 +1,54 @@
+use abstract_dex_adapter::msg::OfferAsset;
+use cosmwasm_std::Addr;
+use cw_storage_plus::{Item, Map};
+
+#[cosmwasm_schema::cw_serde]
+pub struct Config {
+    pub native_denom: String,
+}
+
+#[cosmwasm_schema::cw_serde]
+pub struct ChallengeEntry {
+    pub name: String,
+    pub collateral: Penalty,
+    pub description: String,
+}
+
+#[cosmwasm_schema::cw_serde]
+pub enum Penalty {
+    FixedAmount {
+        asset: OfferAsset,
+    },
+    Daily {
+        asset: OfferAsset,
+        split_between_friends: bool,
+    },
+}
+
+#[cosmwasm_schema::cw_serde]
+pub struct Friend {
+    pub address: Addr,
```

Comment:
Using `Friend` in the public API is dangerous as the `Addr` type assumes the data is already validated. 

Consider changing the field to a generic T: AddressLike and adding a `check` method to `Friend<String>` that returns `Friend<Addr>`.